Code/src/main/java/nl/utwente/group10/haskell/hindley/HindleyMilner.java

```diff
@@ -39,6 +40,8 @@ public static void unify(final Expr context, final Type t1, final Type t2) throw
         final Type a = t1.prune();
         final Type b = t2.prune();
 
+        HindleyMilner.logger.setLevel(Level.OFF);
```

Comment:
It doesn't really make sense to do do this every time unify is called. Maybe you could use a static initializer block here?